src/multibody/geometry.hxx

```diff
@@ -173,100 +217,30 @@ namespace se3
-         it != collisionPairs.end(); ++it)
-      {
-        std::cout << it-> first << "\t" << it->second << std::endl;
-      }
-  }
-
-  inline void GeometryData::activateCollisionPair(const Index pairId,const bool flag)
-  {
-    assert( activeCollisionPairs.size() == model_geom.collisionPairs.size() );
     assert( pairId < activeCollisionPairs.size() );
     activeCollisionPairs[pairId] = flag;
   }
 
-  inline void GeometryData::deactivateCollisionPair(const Index pairId)
+  inline void GeometryData::deactivateCollisionPair(const PairIndex pairId)
   {
-    assert( activeCollisionPairs.size() == model_geom.collisionPairs.size() );
     assert( pairId < activeCollisionPairs.size() );
     activeCollisionPairs[pairId] = false;
   }
 
-  inline bool GeometryData::computeCollision(const Index& pairId)
-  {
-    const CollisionPair & pair = model_geom.collisionPairs[pairId];
-    fcl::CollisionResult& collisionResult = collision_results[pairId];
-
-    const Index & co1 = pair.first;     assert(co1<collisionObjects.size());
-    const Index & co2 = pair.second;    assert(co2<collisionObjects.size());
-
-    collisionResult.clear();
-    fcl::collide (&collisionObjects[co1],&collisionObjects[co2],
-                  collisionRequest, collisionResult);
-
-    return collisionResult.isCollision();
-  }
-  
-  inline void GeometryData::computeAllCollisions()
-  {
-    assert( activeCollisionPairs.size() == model_geom.collisionPairs.size() );
-    assert( collision_results   .size() == model_geom.collisionPairs.size() );
-    for(size_t i = 0; i<model_geom.collisionPairs.size(); ++i)
-    {
-      if(activeCollisionPairs[i])
-        computeCollision(i);
-    }
-  }
-  
-  inline bool GeometryData::isColliding()
-  {
-    Index& i = collisionPairIndex;
-    for(i = 0; i<model_geom.collisionPairs.size(); ++i)
-    {
-      if (activeCollisionPairs[i] && computeCollision(i))
-        return true;
-    }
-    return false;
-  }
-
-  inline DistanceResult GeometryData::computeDistance(const CollisionPair & pair) const
-  {
-    const Index & co1 = pair.first;     assert(co1<collisionObjects.size());
-    const Index & co2 = pair.second;    assert(co2<collisionObjects.size());
-    
-    fcl::DistanceRequest distanceRequest (true, 0, 0, fcl::GST_INDEP);
-    fcl::DistanceResult result;
-    fcl::distance ( &collisionObjects[co1],&collisionObjects[co2],
-                    distanceRequest, result);
-    
-    return DistanceResult (result, co1, co2);
-  }
-  
-  inline void GeometryData::computeAllDistances ()
-  {
-    for(size_t i = 0; i<activeCollisionPairs.size(); ++i)
-    {
-      if (activeCollisionPairs[i])
-        distance_results[i] = computeDistance(model_geom.collisionPairs[i]);
-    }
-  }
-
   inline void GeometryData::resetDistances()
   {
-    std::fill(distance_results.begin(), distance_results.end(), DistanceResult( fcl::DistanceResult(), 0, 0) );
+    std::fill(distanceResults.begin(), distanceResults.end(), fcl::DistanceResult() );
```

Comment:
For collisions, we do the reset during the collision computation (in computeCollision(), collisionResult.clear()).
For distances, we do the reset with this method. 

I guess we should have it with an homogeneous manner. I suggest to remove resetDistances and do the clear in computeDistance(). Would you aggree?